### PR TITLE
Refactor/rubocop line tidy

### PR DESCRIPTION
**Pull request by luke-hill**
##  Summary

Improve Rubocop LineLength (And a few other minor styles)

## Details

Rebase and fix up an old PR fixing up some LineLength issues

## How Has This Been Tested?

CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (cleanup of codebase without changing any existing functionality)
- [ ] Update documentation

## Checklist:

NB: I've not updated the TODO as I want to standardise the TODO first by doing some additional cleanups and removing some patches around certain methods so we have a single source of truth

### CODE REVIEW

File: lib/aruba/api/core.rb [hunk 1]
```diff
@@ -136,7 +138,7 @@ def cd(dir, &block)
       # rubocop:disable Metrics/PerceivedComplexity
       def expand_path(file_name, dir_string = nil)
         # rubocop:disable Layout/LineLength
-        message = %(Filename "#{file_name}" needs to be a string. It cannot be nil or empty either.  Please use `expand_path('.')` if you want the current directory to be expanded.)
+        message = %(Filename "#{file_name}" needs to be a string. It cannot be nil or empty either. Please use `expand_path('.')` if you want the current directory to be expanded.)
```
Comment (mvz) [inline lib/aruba/api/core.rb]: This line is still very long. Please wrap it as well.

File: lib/aruba/config_wrapper.rb [hunk 2]
```diff
@@ -30,7 +30,9 @@ def initialize(config, event_bus)
     # If one method ends with "=", e.g. ":option1=", then notify the event
     # queue, that the user changes the value of "option1"
     def method_missing(name, *args, &block)
-      event_bus.notify Events::ChangedConfiguration.new(changed: { name: name.to_s.gsub(/=$/, ''), value: args.first }) if name.to_s.end_with? '='
+      if name.to_s.end_with? '='
+        event_bus.notify Events::ChangedConfiguration.new(changed: { name: name.to_s.gsub(/=$/, ''), value: args.first })
```
Comment (mvz) [inline lib/aruba/config_wrapper.rb]: This line should be shortened further.

File: lib/aruba/processes/spawn_process.rb [hunk 3]
```diff
@@ -273,7 +273,14 @@ def command_string
...
       def command_path
-        @command_path ||= (Aruba.platform.builtin_shell_commands.include?(command) ? command : Aruba.platform.which(command, environment['PATH']))
+        @command_path ||=
+          begin
+            if Aruba.platform.builtin_shell_commands.include?(command)
+              command
+            else
+              Aruba.platform.which(command, environment['PATH'])
+            end
+          end
```
Comment (mvz) [inline lib/aruba/processes/spawn_process.rb]: :+1:

File: spec/aruba/runtime_spec.rb [hunk 4]
```diff
@@ -1,29 +1,35 @@
...
+    end
+
+    context 'when "/test/fixtures"-directory exist' do
+      it {
+        pending('These tests need fixing and classifying')
+        api.create_directory('test/fixtures')
+        expect(api.fixtures_directory.to_s).to eq expand_path('test/fixtures')
+      }
+    end
+  end
```
Comment (mvz) [inline spec/aruba/runtime_spec.rb]: :+1: Hopefully we can tackle these soon.

File: spec/event_bus_spec.rb [hunk 5]
```diff
@@ -27,10 +27,11 @@ class MyMalformedHandler; end
 
   describe '#notify' do
     before(:each) do
-      allow(name_resolver).to receive(:transform).with(event_name).and_return(event_klass)
+      allow(name_resolver)
+        .to receive(:transform).with(event_name).and_return(event_klass)
     end
 
-    context 'when subscriber to event, the block is called and get\'s an instance of the event passed as payload' do
+    context 'when subscriber to event, the block is called and gets an instance of the event passed as payload' do
```
Comment (mvz) [inline spec/event_bus_spec.rb]: :+1:

File: spec/event_bus_spec.rb [hunk 6]
```diff
@@ -148,11 +153,16 @@ class MyMalformedHandler; end
...
+        expect { bus.register(event_klass, MyMalformedHandler.new) }
+          .to raise_error ArgumentError
+      end
     end
 
     context 'when no handler is given' do
-      it { expect { bus.register(event_klass) }.to raise_error ArgumentError }
+      it 'raises an ArgumentError' do
+        expect { bus.register(event_klass) }.to raise_error ArgumentError
+      end
```
Comment (mvz) [inline spec/event_bus_spec.rb]: Nice improvements to this file!

File: lib/aruba/cucumber/command.rb [hunk 7]
```diff
@@ -136,14 +136,11 @@
...
-              :have_output
-            when :stderr
-              :have_output_on_stderr
-            when :stdout
-              :have_output_on_stdout
+Then(/^(?:the )?(output|stderr|stdout) from "([^"]*)" should contain( exactly)? "([^"]*)"$/) do |channel, cmd, exactly, expected|
+  matcher = case channel
+            when 'output'; then :have_output
+            when 'stderr'; then :have_output_on_stderr
+            when 'stdout'; then :have_output_on_stdout
```
[the remark this answers is not in the document]
Comment (luke-hill) [reply lib/aruba/cucumber/command.rb]: This allows us to clean up a whole load of un-necessary lines.

**Comment by mvz** [comment]
I'll take a look at this again next weekend.

**Comment by luke-hill** [comment]
@mvz I'm aware I need to fix conflicts, but could you re-review this first to work out if I only need to do small things or not?

**Comment by mvz** [comment]
@luke-hill done!

**Comment by mvz** [comment]
@luke-hill are you planning to work on this soon? If not I can fix this up, fix the spec failures, and merge.

**Comment by luke-hill** [comment]
I'll get on this this week

File: lib/aruba/api/core.rb [hunk 8]
```diff
@@ -151,28 +152,28 @@ def expand_path(file_name, dir_string = nil)
...
           path
         elsif prefix == '~'
           path = with_environment do
             ArubaPath.new(File.expand_path(file_name))
           end
 
-          fail ArgumentError, 'Expanding "~/" to "/" is not allowed' if path.to_s == '/'
-          fail ArgumentError, %(Expanding "~/" to a relative path "#{path}" is not allowed) unless path.absolute?
+          raise ArgumentError, 'Expanding "~/" to "/" is not allowed' if path.to_s == '/'
+          raise ArgumentError, "Expanding '~/' to a relative path #{path} is not allowed" unless path.absolute?
```
[the remark this answers is not in the document]
Comment (luke-hill) [reply lib/aruba/api/core.rb]: Can't as there's an interpolation

File: lib/aruba/api/core.rb [hunk 9]
```diff
@@ -152,28 +153,30 @@ def expand_path(file_name, dir_string = nil)
...
           path
         elsif prefix == '~'
           path = with_environment do
             File.expand_path(file_name)
           end
 
-          fail ArgumentError, 'Expanding "~/" to "/" is not allowed' if path == '/'
-          fail ArgumentError, %(Expanding "~/" to a relative path "#{path}" is not allowed) unless Aruba.platform.absolute_path? path
+          raise ArgumentError, 'Expanding "~/" to "/" is not allowed' if path.to_s == '/'
+          raise ArgumentError, "Expanding '~/' to a relative path #{path} is not allowed" unless Aruba.platform.absolute_path? path
```
Comment (mvz) [inline lib/aruba/api/core.rb]: This should work:

```suggestion
          raise ArgumentError, "Expanding \"~\" to a relative path \"#{path}\" is not allowed" unless Aruba.platform.absolute_path? path
```

**Comment by mvz** [comment]
I'm going to rebase and merge this.

**Pull request merged**